core/src/main/java/org/elasticsearch/search/sort/GeoDistanceSortBuilder.java

```diff
@@ -22,33 +22,52 @@
 import org.elasticsearch.ElasticsearchParseException;
 import org.elasticsearch.common.geo.GeoDistance;
 import org.elasticsearch.common.geo.GeoPoint;
+import org.elasticsearch.common.geo.GeoUtils;
+import org.elasticsearch.common.io.stream.NamedWriteable;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.common.unit.DistanceUnit;
+import org.elasticsearch.common.xcontent.ToXContent;
 import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
 import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryParseContext;
+import org.elasticsearch.search.MultiValueMode;
 
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 import java.util.Locale;
+import java.util.Objects;
 
 /**
  * A geo distance based sorting on a geo point like field.
  */
-public class GeoDistanceSortBuilder extends SortBuilder {
+public class GeoDistanceSortBuilder extends SortBuilder implements ToXContent, NamedWriteable<GeoDistanceSortBuilder>, ParameterParser<GeoDistanceSortBuilder> {
+    public static final String NAME = "_geo_distance";
+    public static final boolean DEFAULT_COERCE = false;
+    public static final boolean DEFAULT_IGNORE_MALFORMED = false;
+    public static final String DEFAULT_SORT_MODE = "MAX"; 
+
+    static final GeoDistanceSortBuilder PROTOTYPE = new GeoDistanceSortBuilder("");
 
     final String fieldName;
     private final List<GeoPoint> points = new ArrayList<>();
-    private final List<String> geohashes = new ArrayList<>();
 
-    private GeoDistance geoDistance;
-    private DistanceUnit unit;
-    private SortOrder order;
-    private String sortMode;
+    private GeoDistance geoDistance = GeoDistance.DEFAULT;
+    private DistanceUnit unit = DistanceUnit.DEFAULT;
+    private SortOrder order = SortOrder.DEFAULT;
+    
+    // TODO there is an enum that covers that parameter which we should be using here
+    private String sortMode = DEFAULT_SORT_MODE;
```

Comment:
After going through the change and tests once more: Not setting this to some default, the toXContent/fromXContent roundtrip test fails - in fromXContent the value is being set, if it's set, it's being rendered in toXContent, even if in the initial builder it was neither set nor rendered out. Not sure how to solve this issue.